docker/app/tests/PostControllerTest.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace App\Tests;
+
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+
+use function PHPUnit\Framework\containsEqual;
+use function PHPUnit\Framework\containsOnly;
+use function Symfony\Component\DependencyInjection\Loader\Configurator\env;
+
+class PostControllerTest extends WebTestCase
+{
+    public function testSearchGiftcard(): void
+    {
+        $client = static::createClient();
+        $client->request('GET', '/giftcard/search/d0fea202bab24e60982f');
+        $response = $client->getResponse();
+
+        $this->assertSame(200, $response->getStatusCode());
+        $this->assertJson($response->getContent());
+        #$responseData = json_decode($response->getContent(), true);
+        
+    }
+    public function testCreateGiftcard(): void
+    {
+        
+        $client = static::createClient();
+        
+        $body = [
+                "type" => "DIGITAL",
+                "currency"=>[
+                    "amount"=> 51,
+                    "currency"=> "USD"
+                ],
+                "isValid"=> true
+            ];
+        $body = json_encode($body);
+        $client->request('POST', '/giftcard/create', [], [], [], $body);
+        $response = $client->getResponse();
+        
+        $this->assertSame(200, $response->getStatusCode());
+        $this->assertJson($response->getContent());
+    }
+    public function testInvalidateGiftcard(): void
+    {
+        $array = [];
+        $client = static::createClient();
+        $client->request('PATCH', '/giftcard/invalidate');
```

Comment:
this works because you have a hardcoded value, But you should be able to send ID or unique name of the gift card that needs to be invalidated.